### PR TITLE
Add filters for load deletion

### LINKED ISSUE #567: Missing filters in data processing for gas data
To fullfil the commit https://github.com/openego/eGon-data/pull/478/commits/4b8589ce0dab0df9973b7a27f11fb16ac5dcb8f6, add scanario filter in https://github.com/openego/eGon-data/blob/dev/src/egon/data/datasets/industrial_gas_demand.py.

### LINKED ISSUE #596: Insert carrier information into etrago tables (gas related carriers)
As discussed in #576 the hard coded data in the gas related carrier scripts should be replaced and missing data should be inserted using the `scenario_parameters` data. We will decide whether this is done in a central script or on generation of the data in the respective dataset scripts.

### PR DESCRIPTION
**Pull request by fwitte**
Fixes #567  
Fixes #596

## Before merging into `dev`-branch, please make sure that

- [x] the `CHANGELOG.rst` was updated.
- [x] new and adjusted code is formated using `black` and `isort`.
- [x] the `Dataset`-version is updated when existing datasets are adjusted. 
- [x] the branch was merged into the `continuous-integration/run-everything-over-the-weekend`- branch.
- [x] the workflow is running successful in `test mode`.
- [x] the workflow is running successful in `Everything` mode.

### CODE REVIEW

**Comment by fwitte** [comment]
It seems like we did not delete any data from pypsa-eur-sec with our tasks, can someone verify this? If so, we can merge this into dev.

File: src/egon/data/datasets/gas_grid.py [hunk 1]
```diff
@@ -406,10 +406,16 @@ def insert_gas_pipeline_list(gas_nodes_list):
...
+        f"""SELECT bus_id FROM grid.egon_etrago_bus
+            WHERE scn_name = '{scn_name}' AND country = 'DE';
+        """
+    )['bus_id'])
+
     # Insert data to db
     db.execute_sql(
         f"""DELETE FROM grid.egon_etrago_link WHERE "carrier" = 'CH4' AND
-           scn_name = '{scn_name}';
+           scn_name = '{scn_name}' AND bus0 IN {buses} AND bus1 IN {buses};
```
Comment (ClaraBuettner) [inline src/egon/data/datasets/gas_grid.py:418]: In general you shouldn't exchange data from SQL to python variables just to use them in another SQL query. That results in very long SQL statements which could exceed the limit of allowed characters and needs more computational resources.

File: src/egon/data/datasets/industrial_gas_demand.py [hunk 2]
```diff
@@ -23,23 +23,32 @@ def __init__(self, dependencies):
...
     correspondance_url = (
         "http://opendata.ffe.de:3000/region?id_region_type=eq.38"
     )
     url = "http://opendata.ffe.de:3000/opendata?id_opendata=eq.66&&year=eq."
 
+    if scn_name == 'eGon2035':
+        year = "2035"
+    else:
+        year = "2050"
+
```
Comment (ClaraBuettner) [inline src/egon/data/datasets/industrial_gas_demand.py:50]: You can select the year from scenario_parameters (`get_sector_parameters("global", scn_name)["population_year"]`)

File: src/egon/data/datasets/gas_grid.py (hunk 1 above)
Comment (fwitte) [reply src/egon/data/datasets/gas_grid.py]: Thank you for the hint, changed all queries.

**Comment by fwitte** [comment]
Will merge this into CI again to make sure, the nothing went wrong with the updated SQL queries.

File: src/egon/data/datasets/industrial_gas_demand.py (hunk 2 above)
Comment (ClaraBuettner) [reply src/egon/data/datasets/industrial_gas_demand.py:48]: I run your branch in test mode and it failed due to the datatype of `year`. `get_sector_parameters` returns an `int` value, but in line 51 a `str` value is needed.

Comment (fwitte) [reply src/egon/data/datasets/industrial_gas_demand.py:48]: Thank you for letting me know, fixed by 72cc1f7

**Review by ClaraBuettner — APPROVED** [review]

**Comment by fwitte** [comment]
On successful CI run I will merge #603 into this first and then into dev

**Comment by ClaraBuettner** [comment]
`power_etrago.match_ocgt.insert-open-cycle-gas-turbines` failed on CI:
```
Traceback (most recent call last):
  File "/home/egondata/git-repos/friday-evening-weekend-run/environment/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 984, in _run_raw_task
    result = task_copy.execute(context=context)
  File "/home/egondata/git-repos/friday-evening-weekend-run/code/src/egon/data/datasets/__init__.py", line 195, in skip_task
    result = super(type(task), task).execute(*xs, **ks)
  File "/home/egondata/git-repos/friday-evening-weekend-run/environment/lib/python3.8/site-packages/airflow/operators/python_operator.py", line 113, in execute
    return_value = self.execute_callable()
  File "/home/egondata/git-repos/friday-evening-weekend-run/environment/lib/python3.8/site-packages/airflow/operators/python_operator.py", line 118, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/home/egondata/git-repos/friday-evening-weekend-run/code/src/egon/data/datasets/power_etrago/match_ocgt.py", line 53, in insert_open_cycle_gas_turbines
    gdf["efficiency_fixed"] = scn_params["efficiency"][carrier]
KeyError: 'OCGT'
```
It looks like you have to select the 'gas' parameters for OCGT instead of 'electricity' in  75ca9fc31edba08cbfc23f96a1ef3b933e949a8a

**Comment by fwitte** [comment]
The marginal cost parameter is in the `"electricity"` key and the efficiency parameter in `"gas"`. I think `marginal_cost` for OCGT was included in `"electricity"` before I made those changes, is it okay if I move all parameters of OCGT into the `"gas"` section?

**Pull request merged**